lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -818,26 +819,147 @@ namespace Js
+        JavascriptString *localeJSstr = nullptr;
+
+#if defined(INTL_ICU)
+        // Verify locale is present
+        // REVIEW (doilij): Fix comparison of the unsigned value <= 0
+        if (!GetTypedPropertyBuiltInFrom(options, __locale, JavascriptString) || (localeJSstr = JavascriptString::FromVar(propertyValue))->GetLength() <= 0)
+        {
+            // REVIEW (doilij): We return undefined from all paths here, so should we throw instead to indicate to Intl.js that something went wrong?
+            return scriptContext->GetLibrary()->GetUndefined();
+        }
+
+        // First we have to determine which formatter(number, percent, or currency) we will be using.
+        // Note some options might not be present.
+        PlatformAgnostic::Resource::IPlatformAgnosticResource *numberFormatter = nullptr;
+        const char16 *locale = localeJSstr->GetSz();
+        const charcount_t cch = localeJSstr->GetLength();
+
+        uint16 formatterToUseVal = 0; // 0 (default) is number, 1 is percent, 2 is currency
+        if (GetTypedPropertyBuiltInFrom(options, __formatterToUse, TaggedInt) && (formatterToUseVal = TaggedInt::ToUInt16(propertyValue)) == 1)
+        {
+            // Use the percent formatter (1)
+            IfFailThrowHr(PlatformAgnostic::Intl::CreatePercentFormatter(locale, cch, &numberFormatter));
+        }
+        else if (formatterToUseVal == 2)
+        {
+            // Use the currency formatter (2)
+            if (!GetTypedPropertyBuiltInFrom(options, __currency, JavascriptString))
+            {
+                return scriptContext->GetLibrary()->GetUndefined();
+            }
+
+            JavascriptString *currencyCodeJsString = JavascriptString::FromVar(propertyValue);
+            const char16 *currencyCode = currencyCodeJsString->GetSz();
+
+            // __currencyDisplayToUse: 0 is symbol, 1 is code, 2 is name.
+            if (GetTypedPropertyBuiltInFrom(options, __currencyDisplayToUse, TaggedInt))
+            {
+                uint16 currencyDisplay = TaggedInt::ToUInt16(propertyValue);
+                IfFailThrowHr(PlatformAgnostic::Intl::CreateCurrencyFormatter(locale, cch, currencyCode, currencyDisplay, &numberFormatter));
+            }
+        }
+        else
+        {
+            // Use the number formatter (0 or default)
+            IfFailThrowHr(PlatformAgnostic::Intl::CreateNumberFormatter(locale, cch, &numberFormatter));
+        }
+
+        Assert(numberFormatter);
+        // REVIEW (doilij): AutoPtr will call delete on IPlatformAgnosticResource and complain of non-virtual dtor. There are no IfFailThrowHr so is this still necessary?
+        // TODO (doilij): If necessary, introduce an PlatformAgnosticResourceAutoPtr that calls Cleanup() instead of delete on the pointer.
+        // PlatformAgnostic::Resource::AutoPtr<PlatformAgnostic::Resource::IPlatformAgnosticResource> numberFormatterGuard(numberFormatter);
```

Comment:
Are these comments intentionally left in?

Reply:
For review, yes -- I was hoping to discuss this design with @digitalinfinity and get his sign off before removing the code and comments (or otherwise updating the design).